lib/BackbeatConsumer.js

```diff
@@ -114,6 +144,12 @@ class BackbeatConsumer extends EventEmitter {
             }], () => {
                 this.emit('consumed', count);
                 this._messagesConsumed = 0;
+
+                this.emit('sendMetrics', this._metricsStore, () => {
```

Comment:
I don't like the idea of providing a callback to an event listener, because it's not how the event model fits as they are usually mutually exclusive: normally it should be safe to have 0 or more listeners to any event, here it does seem like the code assumes there will be exactly one listener. In the same vein, you could rename the event to simply `'metrics'`, because emitting events normally does not assume a specific action from the listener(s), rather provides a bit of information that the listener is free to do whatever he needs with.

I may be wrong but it seems you could simply reset the `this._metricsStore` object as soon as you emitted the `sendMetrics` event, because the original object will be left untouched for event listeners.